pkg/model/iam.go

```diff
@@ -64,33 +65,52 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	// Generate IAM objects etc for each role
 	for _, role := range roles {
 		name := b.IAMName(role)
+		roleAsString := reflect.ValueOf(role).String()
 
 		var iamRole *awstasks.IAMRole
-		{
-			rolePolicy, err := b.buildAWSIAMRolePolicy()
-			if err != nil {
-				return err
-			}
 
+		customIamRoles := b.Cluster.Spec.RoleCustomIamRoles
+
+		// If we've specified an IAMRoleArn for this cluster role,
+		// do not create a new one
+		if customIamRoles[roleAsString] != "" {
```

Comment:
**Review Notes**

I do need to have you put this functionality behind a feature flag.

- Here is our implementation of feature flags: https://github.com/kubernetes/kops/blob/77fbf9cbf9003d89d2e157027ac1e650c1798c60/pkg/featureflag/featureflag.go
- here is the documentation about one https://github.com/kubernetes/kops/blob/77fbf9cbf9003d89d2e157027ac1e650c1798c60/cmd/kops/rollingupdatecluster.go#L105
- here is an example of use https://github.com/kubernetes/kops/blob/77fbf9cbf9003d89d2e157027ac1e650c1798c60/cmd/kops/rollingupdatecluster.go#L119